### ISSUE TITLE
[BUG] clip_grad_norm for zero_optimization mode is not working

### ISSUE THREAD

**Issue by chengmengli06** [microsoft/DeepSpeed#6767]
set "gradient_clipping" in deepspeed does not work, look into the source code in deepspeed.runtime.engine.DeepSpeedEngine,in line 2101
```
    def _take_model_step(self, lr_kwargs, block_eigenvalue={}):
        if self.gradient_clipping() > 0.0:
            if not (self.fp16_enabled() or self.bfloat16_enabled() or self.amp_enabled() or self.zero_optimization()):
                self.clip_fp32_gradients()
            elif self.amp_enabled():
                # AMP's recommended way of doing clipping
                # https://nvidia.github.io/apex/advanced.html#gradient-clipping
                master_params = amp.master_params(self.optimizer)
                clip_grad_norm_(parameters=master_params, max_norm=self.gradient_clipping(), mpu=self.mpu)
        self.optimizer.step()
```
thus gradient clipping do nothing at all!!!

**Comment by tjruwase** [comment]
@chengmengli06, this is incorrect reading of the code. Gradient clipping is handled in the respective optimizer implementations such as:

1. [bf16 optim](https://github.com/microsoft/DeepSpeed/blob/83e4364fbd921ad4fe808d31eb107ef080228a7a/deepspeed/runtime/bf16_optimizer.py#L290)
2. [fp16 optim](https://github.com/microsoft/DeepSpeed/blob/83e4364fbd921ad4fe808d31eb107ef080228a7a/deepspeed/runtime/fp16/fused_optimizer.py#L358)
3. [zero](https://github.com/microsoft/DeepSpeed/blob/83e4364fbd921ad4fe808d31eb107ef080228a7a/deepspeed/runtime/zero/stage_1_and_2.py#L1964)

**Comment by chengmengli06** [comment]
I find it, and verify that it does work under zero_2 mode. Thanks!
